### ISSUE TITLE
Calendar: "Error: Duplicate global Esc key listener with priority [600, 0] in Calendar components"

### ISSUE THREAD

**Issue by Oko-Tester** [primefaces/primereact#7407]
### Describe the bug

The following error occurs when interacting with multiple calendar components in PrimeReact:

```
Uncaught Error: Unexpected: global esc key listener with priority [600, 0] already exists.
    at Object.addListener (_app-076b402d275c7b07.js:1:150514)
    at _app-076b402d275c7b07.js:1:150856
    at ti (framework-eae1dab60e375b5f.js:1:93682)
    at ws (framework-eae1dab60e375b5f.js:1:113909)
    at framework-eae1dab60e375b5f.js:1:112644
    at ks (framework-eae1dab60e375b5f.js:1:112710)
    at us (framework-eae1dab60e375b5f.js:1:106429)
    at jl (framework-eae1dab60e375b5f.js:1:46917)
    at framework-eae1dab60e375b5f.js:1:103824
```
Error also occurs in the official PrimeReact documentation.

### Reproducer

https://stackblitz.com/edit/rtiuwb?file=src%2FApp.jsx

### System Information

```Shell
npmPackages:
   primereact: ^10.8.3 => 10.8.3
   react: ^18.3.1 => 18.3.1
```

### Steps to reproduce the behavior

1. Click on a calendar component and select a date.
2. Click on a different calendar component to open it.
3. The error appears in the console.

### Expected behavior

The global listener for the Escape key should not be added multiple times. Ideally, a single listener should be sufficient, or any existing listener should be removed before adding a new one.

**Comment by melloware** [comment]
@Oko-Tester this was fixed in 10.8.4: https://github.com/primefaces/primereact/issues/7220  can you try 10.8.4?

**Comment by Oko-Tester** [comment]
@melloware I updated my version, and it works now. Sorry for the inconvenience.
